### PR TITLE
[SRP] 2022-05-01 Swagger Api

### PR DESCRIPTION
**Pull request by MaximeKjaer**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
        
        `"directoryServiceOptions": "AADKERB"` is already deployed (see [public preview instructions](https://docs.microsoft.com/en-us/azure/virtual-desktop/create-profile-container-azure-ad))
   
  4. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
      
      ASAP.
      
  6. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by MaximeKjaer** [comment]
Thanks @jianyexi, I've reviewed and checked the box.

Do you perhaps know what I can do to make the checks run? They've been queued for 20 hours, and haven't run yet on my latest commit.

**Review by raosuhas — APPROVED** [review]
:shipit:

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]
Please fix the [R2018 - XmsEnumValidation](https://github.com/Azure/azure-rest-api-specs/blob/main/documentation/openapi-authoring-automated-guidelines.md#R2018) alarms reported in LintDiff check. Those errors are diffed, but it does not mean we do not need them fixed. Missing of these `x-ms-enum` usually generates into user-unfriendly SDKs.

**Comment by MaximeKjaer** [comment]
Thanks @ArcturusZhang. Fixed the `XmsEnumValidation` lint errors.

@HimanshuChhabra I marked them all with `"modelAsString": true`. That seems to be the more common thing to do, and it gives us flexibility to add values to the enum later on, without having to make a breaking change. But let me know if you would prefer some of them to be `"modelAsString": false`.

@ArcturusZhang I see there are other failures, notably:

```
cmderr	[generator automation-v2] [ERROR] fatal   | Error: Enum types of "array" and format "undefined" are not supported. Correct your input (AllowedMethods).
```

Is this also something that needs to be fixed? If so, do you know what might be causing this error?

File: specification/storage/resource-manager/Microsoft.Storage/stable/2022-05-01/common.json
Comment (blueww) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2022-05-01/common.json:47]: Generate SDK meet following error, "x-ms-enum" should be in same level as "enum":

```
FATAL: System.InvalidOperationException: Found 'x-ms-enum' without 'enum' on the same level. Please either add an 'enum' restriction or remove 'x-ms-enum'.
   at AutoRest.Modeler.ObjectBuilder.BuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/ObjectBuilder.cs:line 63
   at AutoRest.Modeler.SchemaBuilder.ParentBuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/SchemaBuilder.cs:line 217
   at AutoRest.Modeler.SchemaBuilder.BuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/SchemaBuilder.cs:line 48
   at AutoRest.Modeler.SchemaBuilder.BuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/SchemaBuilder.cs:line 133
   at AutoRest.Modeler.ObjectBuilder.BuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/ObjectBuilder.cs:line 182
   at AutoRest.Modeler.SchemaBuilder.ParentBuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/SchemaBuilder.cs:line 217
   at AutoRest.Modeler.SchemaBuilder.BuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/SchemaBuilder.cs:line 48
   at AutoRest.Modeler.SchemaBuilder.BuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/SchemaBuilder.cs:line 133
   at AutoRest.Modeler.SchemaBuilder.BuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/SchemaBuilder.cs:line 133
   at AutoRest.Modeler.SchemaBuilder.BuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/SchemaBuilder.cs:line 133
   at AutoRest.Modeler.SwaggerModeler.BuildCompositeTypes() in /opt/vsts/work/1/s/src/SwaggerModeler.cs:line 348
   at AutoRest.Modeler.SwaggerModeler.Build(ServiceDefinition serviceDefinition) in /opt/vsts/work/1/s/src/SwaggerModeler.cs:line 66
   at AutoRest.Modeler.Program.<ProcessInternal>d2.MoveNext() in /opt/vsts/work/1/s/src/Program.cs:line 60
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NewPlugin.<Process>d15.MoveNext()
```

**Comment by blueww** [comment]
It looks server still not ready for the new API version. 
We should not merge the PR before the new api version is generally avaialbe in all regions.
I have tested stage, canary, westus, and all failed with error like following:
```
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
GET

Absolute Uri:
https://management.azure.com/subscriptions/***/resourceGroups/***/providers/Microsoft.Storage/storageAccounts/***?api-version=2022-05-01

Headers:
x-ms-client-request-id        : 1af59fad-7da0-4724-ad7c-9cfff8831360
accept-language               : en-US

Body:



DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
BadRequest

Headers:
Pragma                        : no-cache
x-ms-failure-cause            : gateway
x-ms-request-id               : da8afbeb-6961-4c5c-9021-e4c406bf690a
x-ms-correlation-request-id   : da8afbeb-6961-4c5c-9021-e4c406bf690a
x-ms-routing-request-id       : SOUTHEASTASIA:20220613T133946Z:da8afbeb-6961-4c5c-9021-e4c406bf690a
Strict-Transport-Security     : max-age=31536000; includeSubDomains
X-Content-Type-Options        : nosniff
Cache-Control                 : no-cache
Date                          : Mon, 13 Jun 2022 13:39:46 GMT

Body:
{
  "error": {
    "code": "NoRegisteredProviderFound",
    "message": "No registered resource provider found for location 'eastus2stage' and API version '2022-05-01' for type 'storageAccounts'. The supported api-versions are '2021-09-01, 2021-08-01, 2021-06-01, 2021-05-01, 2021-04-01, 2021-02-01, 202
1-01-01, 2020-08-01-preview, 2019-06-01, 2019-04-01, 2018-11-01, 2018-07-01, 2018-03-01-preview, 2018-02-01, 2017-10-01, 2017-06-01, 2016-12-01, 2016-05-01, 2016-01-01, 2015-06-15, 2015-05-01-preview'. The supported locations are 'eastus, eastus2
, westus, westeurope, eastasia, southeastasia, japaneast, japanwest, northcentralus, southcentralus, centralus, northeurope, brazilsouth, australiaeast, australiasoutheast, southindia, centralindia, westindia, canadaeast, canadacentral, westus2, 
westcentralus, uksouth, ukwest, koreacentral, koreasouth, francecentral, australiacentral, southafricanorth, uaenorth, switzerlandnorth, germanywestcentral, norwayeast, westus3, jioindiawest, swedencentral, eastus2stage, eastus2euap, centraluseua
p'."
  }
}
```

**Review by blueww — CHANGES_REQUESTED** [review]
It looks server still not ready for the new API version. 
We should not merge the PR before the new api version is generally avaialbe in all regions.
I have tested stage, canary, westus, and all failed with error like following:
```
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
GET

Absolute Uri:
https://management.azure.com/subscriptions/***/resourceGroups/***/providers/Microsoft.Storage/storageAccounts/***?api-version=2022-05-01

Headers:
x-ms-client-request-id        : 1af59fad-7da0-4724-ad7c-9cfff8831360
accept-language               : en-US

Body:



DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
BadRequest

Headers:
Pragma                        : no-cache
x-ms-failure-cause            : gateway
x-ms-request-id               : da8afbeb-6961-4c5c-9021-e4c406bf690a
x-ms-correlation-request-id   : da8afbeb-6961-4c5c-9021-e4c406bf690a
x-ms-routing-request-id       : SOUTHEASTASIA:20220613T133946Z:da8afbeb-6961-4c5c-9021-e4c406bf690a
Strict-Transport-Security     : max-age=31536000; includeSubDomains
X-Content-Type-Options        : nosniff
Cache-Control                 : no-cache
Date                          : Mon, 13 Jun 2022 13:39:46 GMT

Body:
{
  "error": {
    "code": "NoRegisteredProviderFound",
    "message": "No registered resource provider found for location 'eastus2stage' and API version '2022-05-01' for type 'storageAccounts'. The supported api-versions are '2021-09-01, 2021-08-01, 2021-06-01, 2021-05-01, 2021-04-01, 2021-02-01, 202
1-01-01, 2020-08-01-preview, 2019-06-01, 2019-04-01, 2018-11-01, 2018-07-01, 2018-03-01-preview, 2018-02-01, 2017-10-01, 2017-06-01, 2016-12-01, 2016-05-01, 2016-01-01, 2015-06-15, 2015-05-01-preview'. The supported locations are 'eastus, eastus2
, westus, westeurope, eastasia, southeastasia, japaneast, japanwest, northcentralus, southcentralus, centralus, northeurope, brazilsouth, australiaeast, australiasoutheast, southindia, centralindia, westindia, canadaeast, canadacentral, westus2, 
westcentralus, uksouth, ukwest, koreacentral, koreasouth, francecentral, australiacentral, southafricanorth, uaenorth, switzerlandnorth, germanywestcentral, norwayeast, westus3, jioindiawest, swedencentral, eastus2stage, eastus2euap, centraluseua
p'."
  }
}
```

**Comment by jianyexi** [comment]
@ArcturusZhang  @qiaozha  could you help evaluate if we need involve breaking change review board to review , as there are SDK breaking change?

File: specification/storage/resource-manager/Microsoft.Storage/stable/2022-05-01/common.json
Comment (MaximeKjaer) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2022-05-01/common.json]: Thanks -- fixed !

**Comment by MaximeKjaer** [comment]
I do see that the tooling is reporting breaking changes, but that is not the intention. The changes in this version are intended to be:

- Adding an enum value to `directoryServiceOptions`
- Making some fields of `activeDirectoryProperties` optional
- Making the `keySource` field optional

I'll look into what the build is highlighting as a breaking change

**Comment by MaximeKjaer** [comment]
I see the breaking changes are reported in previous versions, e.g. `2021-04-01` or `2019-06-01`. Other build failures also seem to be related to breaking changes that have been made in past versions.

Did I perhaps misconfigure the build in a way that causes the CI to check for breaking changes in the full history of the module, instead of just in the most recent version? Or is this the expected behavior of the CI?

This PR is only about making a few non-breaking changes compared to the previous version, so I was expecting it to clear the CI

**Comment by jianyexi** [comment]
@msyyc @ArcturusZhang  @qiaozha could you confirm the SDK breaking change ?

**Comment by msyyc** [comment]
@MaximeKjaer  @jianyexi  we are working on upgrading autorest.python version to fix the ci failure. It may take some time so please ignore the python failure for this PR.

**Review by blueww — APPROVED** [review]

**Comment by MaximeKjaer** [comment]
@ArcturusZhang I addressed your change request for the missing `x-ms-enum`, but I believe we need you to approve that it's been addressed appropriately before we can merge.

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**
